spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/StreamEmitterAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,279 @@
+		}
+
+		if (!Void.TYPE.equals(method.getReturnType())) {
+			if (targetBean == null) {
+				targetBean = this.applicationContext.getBean(outboundName);
+			}
+			boolean streamListenerResultAdapterFound = false;
+			for (StreamListenerResultAdapter streamListenerResultAdapter : this.streamListenerResultAdapters) {
+				if (streamListenerResultAdapter.supports(result.getClass(), targetBean.getClass())) {
+					Object adapted = streamListenerResultAdapter.adapt(result, targetBean);
+					if (adapted != null && Disposable.class.isAssignableFrom(adapted.getClass())) {
+						fluxDisposables.add((Disposable) adapted);
+					}
+					streamListenerResultAdapterFound = true;
+					break;
+				}
+			}
+			Assert.state(streamListenerResultAdapterFound, StreamEmitterErrorMessages.CANNOT_CONVERT_RETURN_TYPE_TO_ANY_AVAILABLE_RESULT_ADAPTERS);
+		}
+	}
+
+	private static void validateStreamEmitterMethod(Method method, int outputAnnotationCount, String methodAnnotatedOutboundName) {
+		if (StringUtils.hasText(methodAnnotatedOutboundName)) {
+			Assert.isTrue(outputAnnotationCount == 0,
+					StreamEmitterErrorMessages.INVALID_OUTPUT_METHOD_PARAMETERS);
+		}
+		else {
+			Assert.isTrue(outputAnnotationCount > 0, StreamEmitterErrorMessages.NO_OUTPUT_SPECIFIED);
+		}
+
+		if (!method.getReturnType().equals(Void.TYPE)) {
+			Assert.isTrue(StringUtils.hasText(methodAnnotatedOutboundName),
+					StreamEmitterErrorMessages.RETURN_TYPE_NO_OUTBOUND_SPECIFIED);
+			Assert.isTrue(method.getParameterCount() == 0, StreamEmitterErrorMessages.RETURN_TYPE_METHOD_ARGUMENTS);
+		}
+		else {
+			if (!StringUtils.hasText(methodAnnotatedOutboundName)) {
+				int methodArgumentsLength = method.getParameterTypes().length;
+				for (int parameterIndex = 0; parameterIndex < methodArgumentsLength; parameterIndex++) {
+					MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method, parameterIndex);
+					if (methodParameter.hasParameterAnnotation(Output.class)) {
+						String outboundName = (String) AnnotationUtils
+								.getValue(methodParameter.getParameterAnnotation(Output.class));
+						Assert.isTrue(StringUtils.hasText(outboundName), StreamEmitterErrorMessages.INVALID_OUTBOUND_NAME);
+					}
+					else {
+						throw new IllegalArgumentException(
+								StreamEmitterErrorMessages.OUTPUT_ANNOTATION_MISSING_ON_METHOD_PARAMETERS_VOID_RETURN_TYPE);
+					}
+				}
+			}
+		}
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return true;
+	}
+
+	@Override
+	public void stop(Runnable callback) {
+		stop();
+		if (callback != null) {
+			callback.run();
+		}
+	}
+
+	@Override
+	public void stop() {
+		try {
+			lock.lock();
+			for (FluxSender fluxSender : fluxSenders) {
+				fluxSender.close();
+			}
+			for (Disposable disposable : fluxDisposables) {
+				disposable.dispose();
+			}
+			if (running) {
+				this.running = false;
+			}
+		}
+		catch (IOException e) {
+			log.error("Error closing reactive source", e);
```

Comment:
Shouldn't we consider `try...catch` individually?
Why an error on one `FluxSender.close()` should prevent all others to be closed afterward?